MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -119,19 +125,141 @@ CompletionContext CreateCompletionContext (List<CompletionItem> items)
 
 			foreach (var att in rr.GetAttributeCompletions (doc, doc.ToolsVersion)) {
 				if (!existingAtts.ContainsKey (att.Name)) {
-					items.Add (CreateCompletionItem (att));
+					items.Add (CreateCompletionItem (att, MSBuildSpecialCommitKind.Attribute));
 				}
 			}
 
 			return CreateCompletionContext (items);
 		}
 
-		CompletionItem CreateCompletionItem (BaseInfo info, string prefix = null)
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
+			IAsyncCompletionSession session,
+			SnapshotPoint triggerLocation,
+			List<XObject> nodePath,
+			IAttributedXObject attributedObject,
+			XAttribute attribute,
+			CancellationToken token)
+		{
+			/*if (!attribute.NameEquals ("include", true)) {
+				return CompletionContext.Empty;
+			}*/
+
+			var context = await GetSessionContext (session, triggerLocation, token);
+			var rr = context.rr;
+			var doc = context.doc;
+
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			if (attributedObject is XElement xElement &&
+				xElement.NameEquals ("PackageReference", true)) {
+
+				switch (Regex.Replace (attribute.FriendlyPathRepresentation.ToLower (), @"[^0-9a-zA-Z]+", "")) {
+				case "include":
+					return await GetPackageNamesAsync (doc, rr);
+
+				case "version":
+					return await GetPackageVersionCompletions (doc, rr, token);//GetPackageVersionsAsync (doc, rr);
+				}
+			}
+
+			return CompletionContext.Empty;
+		}
+
+		async Task<CompletionContext> GetPackageNamesAsync(MSBuildRootDocument doc,
+														MSBuildResolveResult rr)
+        {
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			var items = new List<CompletionItem> ();
+			var httpClient = new HttpClient ();
+			string result = null;
+			var searchQuery = rr.Reference.ToString ().ToLower ();
```

Comment:
Why do you need `ToLower()` here?

Reply:
Wanted to make sure that searchQuery for the API call is not case sensitive. Generally, it doesn't matter as the API call is not case sensitive. But I put it there just to make sure that no Nuget Package is missed because of any case sensitivity issue.